### PR TITLE
[dashboards] New, export api

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Implements a dashboard export API endpoint. To have the exact same functionality on the API and on MVC `ModelView`. This will be used by the new React view for dashboards. 

Added enhanced security for the export, applying the filter that's applied for list and get

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

@nytai

### CODE REVIEW

**Review by nytai — APPROVED** [review]
lgtm, thanks for implementing this :)

File: superset/views/dashboard/api.py [hunk 1]
```diff
@@ -169,6 +173,7 @@ class DashboardRestApi(DashboardMixin, BaseSupersetModelRestApi):
     method_permission_name = {
         "get_list": "list",
         "get": "show",
+        "export": "list",
```
Comment (dpgaspar) [inline superset/views/dashboard/api.py]: Switched export to a `list` permission to reduce the amount of permissions

Comment (nytai) [reply superset/views/dashboard/api.py]: 👍 makes sense as it's essentially another form of list

File: tests/dashboard_api_tests.py [hunk 2]
```diff
@@ -423,3 +423,40 @@ def test_get_related_fail(self):
...
+        """
+            Dashboard API: Test dashboard export
+        """
+        self.login(username="admin")
+        argument = [1, 2]
+        uri = f"api/v1/dashboard/export/?q={prison.dumps(argument)}"
+
+        rv = self.client.get(uri)
+        self.assertEqual(rv.status_code, 200)
+        rv.headers["Content-Disposition"] = 'attachment; filename="dashboards.json"'
```
Comment (suddjian) [inline tests/dashboard_api_tests.py]: Is this line supposed to be asserting something?

File: superset/views/dashboard/api.py (hunk 1 above)
Comment (robdiciuccio) [reply superset/views/dashboard/api.py]: Seems like a different level of permission if the export also contains all of the chart and data source info. Was this the `can_download_dashboards` permission? Can we use the same?

File: superset/views/dashboard/api.py [hunk 3]
```diff
@@ -356,3 +361,52 @@ def delete(self, item):  # pylint: disable=arguments-differ
...
+              $ref: '#/components/responses/500'
+        """
+        query = self.datamodel.session.query(Dashboard).filter(
+            Dashboard.id.in_(kwargs["rison"])
+        )
+        query = self._base_filters.apply_all(query)
+        ids = [item.id for item in query.all()]
+        if not ids:
+            return self.response_404()
+        export = Dashboard.export_dashboards(kwargs["rison"])
```
Comment (robdiciuccio) [inline superset/views/dashboard/api.py]: `Dashboard.export_dashboards(ids)`?

File: superset/views/dashboard/views.py [hunk 4]
```diff
@@ -44,8 +44,7 @@ class DashboardModelView(
     datamodel = SQLAInterface(models.Dashboard)
 
     @action("mulexport", __("Export"), __("Export dashboards?"), "fa-database")
-    @staticmethod
-    def mulexport(items):
+    def mulexport(self, items):  # pylint: disable=no-self-use
```
Comment (robdiciuccio) [inline superset/views/dashboard/views.py]: What's the reasoning behind this change?

File: tests/dashboard_api_tests.py [hunk 5]
```diff
@@ -423,3 +423,40 @@ def test_get_related_fail(self):
...
         rv = self.client.get(uri)
         self.assertEqual(rv.status_code, 404)
+
+    def test_export(self):
+        """
+            Dashboard API: Test dashboard export
+        """
+        self.login(username="admin")
+        argument = [1, 2]
+        uri = f"api/v1/dashboard/export/?q={prison.dumps(argument)}"
```
Comment (robdiciuccio) [inline tests/dashboard_api_tests.py]: So the final URL here looks like `api/v1/dashboard/export/?q=!(1,2)`?

File: superset/views/dashboard/views.py (hunk 4 above)
Comment (dpgaspar) [reply superset/views/dashboard/views.py]: It's related with this already merged PR: https://github.com/apache/incubator-superset/pull/8942

the `@action` decorator creates a methods attribute on the method, this is defeated by making the method static

File: tests/dashboard_api_tests.py (hunk 5 above)
Comment (dpgaspar) [reply tests/dashboard_api_tests.py]: exactly

File: superset/views/dashboard/api.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/dashboard/api.py]: On this case is `mulexport` changed it

File: tests/dashboard_api_tests.py (hunk 2 above)
Comment (dpgaspar) [reply tests/dashboard_api_tests.py]: No, but now it is

**Review by willbarrett — APPROVED** [review]
LGTM!

**Review by suddjian — APPROVED** [review]
yas

**Review by robdiciuccio — APPROVED** [review]

**Pull request merged**